test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,618 @@
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'RichText', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should handle change in tag name gracefully', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression test: The heading block changes the tag name of its
+		// RichText element. Historically this has been prone to breakage,
+		// because the Editable component prevents rerenders, so React cannot
+		// update the element by itself.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/3091
+		await editor.insertBlock( { name: 'core/heading' } );
+
+		await page.click( '[aria-label="Change heading level"]' );
+		await page.click( 'role=menuitemradio[name="Heading 3"i]' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should apply formatting with primary shortcut', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'test' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should apply formatting when selection is collapsed', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Some ' );
+		// All following characters should now be bold.
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( 'bold' );
+		// All following characters should no longer be bold.
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should apply multiple formats when selection is collapsed', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'i' );
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'i' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should not highlight more than one format', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( ' 2' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowLeft' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+
+		const count = await page.evaluate(
+			() =>
+				document.querySelectorAll( '*[data-rich-text-format-boundary]' )
+					.length
+		);
```

Comment:
Would this also work using `locator.count()`? For Example:

```js
const count = await page.locator( '*[data-rich-text-format-boundary]' ).count();
```

Reply:
Good catch. Thank you @WunderBart.

Reply:
Good catch. Thank you @WunderBart.